src/Foundation/NSArray.cs

```diff
@@ -395,6 +408,18 @@ internal static NativeHandle GetAtIndex (NativeHandle handle, nuint i)
 			return Runtime.GetINativeObject<T> (val, false);
 		}
 
+		static object UnsafeGetItem (NativeHandle handle, nuint index, Type type)
+		{
+			var val = GetAtIndex (handle, index);
+			// A native code could return NSArray with NSNull.Null elements
+			// and they should be valid for things like T : NSDate so we handle
+			// them as just null values inside the array
+			if (val == NSNull.Null.Handle)
+				return null;
+
+			return Runtime.GetINativeObject (val, false, type);
```

Comment:
very nit but could be simplified with a ternary? 
`return val == NSNull.Null.Handle ?  null : Runtime.GetINativeObject (val, false, type);`

Reply:
Yes, that would be equivalent.

It's really a matter of personal taste though, I tend to prefer less code on each line.

One reason is that it's easier to add debug statements:

```cs
if (val == NSNull.Null.Handle) {
    Console.WriteLine ("Nope, that failed");
    return null;
}

Console.WriteLine ("This worked!");
return Runtime.GetINativeObject (val, false, type);
```

but ultimately it doesn't matter either way.

Reply:
Readability should over big one liners 👍